reviews/models/mixins.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from transitions import Machine
+
+from django.db import models
+from django.db import transaction
+from django.utils import timezone
+
+from osf.models.action import Action
+from reviews import workflow
+from reviews.exceptions import InvalidTransitionError
+
+
+class ReviewProviderMixin(models.Model):
+    """A reviewed/moderated collection of objects.
+    """
+
+    REVIEWABLE_RELATION_NAME = None
+
+    class Meta:
+        abstract = True
+
+    reviews_workflow = models.CharField(null=True, blank=True, max_length=15, choices=workflow.Workflows.choices())
+    reviews_comments_private = models.NullBooleanField()
+    reviews_comments_anonymous = models.NullBooleanField()
+
+    @property
+    def is_moderated(self):
+        return self.reviews_workflow is not None
+
+    def get_reviewable_status_counts(self):
+        assert self.REVIEWABLE_RELATION_NAME, 'REVIEWABLE_RELATION_NAME must be set to compute status counts'
+        # TODO fix hackery once GUID query set is gone
```

Comment:
GuidMixinQuerySet has changed dramatically recently. Is this still an issue?

If the issue is the auto-included GUIDs, then you can remove the `include` using `.include(None)`.

Something like

```
base_qs = getattr(self, self.REVIEWABLE_RELATION_NAME)
if isinstance(base_qs, GuidMixinQueryset):
    base_qs = base_qs.include(None)
qs = base_qs.values('....')
```

Reply:
Yeah, that works nicely, thanks